src/bosh-template/spec/unit/bosh/template/evaluation_context_spec.rb

```diff
@@ -109,7 +110,86 @@ def eval_template(erb, context)
       end
 
       let(:evaluation_context) do
-        EvaluationContext.new(spec, dns_encoder)
+        EvaluationContext.new(Bosh::Common::DeepCopy.copy(spec), dns_encoder)
+      end
+
+      context 'operator ==' do
+        let(:other_evaluation_context) do
+          EvaluationContext.new(Bosh::Common::DeepCopy.copy(spec), dns_encoder)
+        end
+
+        context 'when nothing changes' do
+          it 'returns true' do
+            expect(evaluation_context == other_evaluation_context).to equal(true)
+          end
+        end
+
+        context 'when spec changes' do
+          it 'returns false' do
+            evaluation_context.spec['job']['name'] = 'modified_job_name'
+            expect(evaluation_context == other_evaluation_context).to equal(false)
+          end
+        end
+
+        context 'when properties changes' do
+          it 'returns false' do
+            evaluation_context.properties.foo = 'modified_bar'
+            expect(evaluation_context == other_evaluation_context).to equal(false)
+          end
+        end
+
+        context 'when raw_properties changes' do
+          it 'returns false' do
+            evaluation_context.raw_properties['foo'] = 'modified_bar'
+            expect(evaluation_context == other_evaluation_context).to equal(false)
+          end
+        end
+
+        context 'when name changes' do
+          it 'returns false' do
+            evaluation_context.name << 'modified_name'
+            expect(evaluation_context == other_evaluation_context).to equal(false)
+          end
+        end
+
+        context 'when index changes' do
+          module MakeIndexAccessible
+            refine Bosh::Template::EvaluationContext do
+              def modify_index
+                @index = 42
+              end
+            end
+          end
+          using MakeIndexAccessible
+
+          it 'returns false' do
+            evaluation_context.modify_index
+            expect(evaluation_context.index).to equal(42)
+            expect(evaluation_context == other_evaluation_context).to equal(false)
+          end
+        end
+
+        context 'when instance variables are modified' do
```

Comment:
@dpb587-pivotal @charleshansen 
We implemented a [test|] which will check whether a new member is added to the EvaluationContext class which is not yet part of `EvaluationContext.==`. With this kind of test, members which are intentionally not part of `EvaluationContext.==` need to be listed explicitly in the test. We think the test provides additional value but are unsure if it is worth the added complexity. What is your opinion?

Reply:
Verbose, but dynamically defensive; clever. I'm okay with it.